### PR TITLE
Documentation skeleton for the distributed James server

### PR DESCRIPTION
**Pull request by chibenwa**
Draft.

Review welcomed on the overall structure, the few parts of the skeleton I detailed, the TODO you would like to see added.

Grammar reviewer would be welcome to propose easy-to apply Github code suggestions for the many typos I indubitably did...

Note: given the time available, I did not succeeded to build this. There is likely some errors, that I will correct later on.

### CODE REVIEW

File: docs/modules/servers/pages/distributed.adoc [hunk 1]
```diff
@@ -1,5 +1,21 @@
...
 
-(TODO)
+The Distributed server offers an easy to scale email server. Based on
+standard noSQL solutions, it is intended to be easy to operate for large
+deployments, without needing sharding and protocol aware load balancing.
 
+This server is:
+
+* Intended for use by experts only
+* Use for large-scale distributed deployments
```
Comment (Arsnael) [inline docs/modules/servers/pages/distributed.adoc:11]:
```suggestion
* Used for large-scale distributed deployments
```

File: docs/modules/servers/pages/distributed/architecture.adoc [hunk 2]
```diff
@@ -0,0 +1,16 @@
+= Distributed Server Architecture
+
+This sections presents the Distributed Server architecture.
+
+In order to deliver its promises, the Distributed Server leverages the following storage strategies:
+
+(TODO picture)
+
+ * *Cassandra* is used for metadata storage
+ * The *blob store* storage interfaces is responsible of storing potentially large binary data. For instance
```
Comment (Arsnael) [inline docs/modules/servers/pages/distributed/architecture.adoc:10]:
```suggestion
 * The *blob store* storage interface is responsible of storing potentially large binary data. For instance
```

File: docs/modules/servers/pages/distributed/architecture.adoc [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+ email bodies, headers or attachments. Different technologies can be used: *Cassandra*, or *Object Storage*
+(S3 or Swift)
+ * *ElasticSearch* component enpower full text search on emails.
```
Comment (Arsnael) [inline docs/modules/servers/pages/distributed/architecture.adoc:13]:
```suggestion
 * *ElasticSearch* component empowers full text search on emails.
```

File: docs/modules/servers/pages/distributed/extend/index.adoc [hunk 4]
```diff
@@ -0,0 +1,18 @@
+= Extending the behavior of the Distributed Server
+
+The Distributed Server exposes several interfaces allowing the user to write custom extensions in
+order to extend the Distributed Server behavior.
+
+Writing *Mailet* and *Matchers* allows one to supply custom components for the
```
Comment (Arsnael) [inline docs/modules/servers/pages/distributed/extend/index.adoc:6]:
```suggestion
Writing *Mailets* and *Matchers* allows one to supply custom components for the
```

File: docs/modules/servers/pages/distributed/operate/index.adoc [hunk 5]
```diff
@@ -0,0 +1,20 @@
+= Operate the Distributed server
+
+The following pages details how to operate the Distributed server.
```
Comment (Arsnael) [inline docs/modules/servers/pages/distributed/operate/index.adoc:3]:
```suggestion
The following pages detail how to operate the Distributed server.
```

File: docs/modules/servers/pages/distributed.adoc [hunk 6]
```diff
@@ -1,5 +1,21 @@
 = James Distributed Mail Server
 :navtitle: Distributed
 
-(TODO)
+The Distributed server offers an easy to scale email server. Based on
```
Comment (rouazana) [inline docs/modules/servers/pages/distributed.adoc:4]:
```suggestion
The Distributed server offers an easy way to scale email server. Based on
```

File: docs/modules/servers/pages/distributed.adoc [hunk 7]
```diff
@@ -1,5 +1,21 @@
...
+standard noSQL solutions, it is intended to be easy to operate for large
```
Comment (rouazana) [inline docs/modules/servers/pages/distributed.adoc:5]: standard and noSQL sounds weird :)

**Comment by chibenwa** [comment]
> are you aware that you can apply all suggestions in only one commit?

No I was not.

**Comment by chibenwa** [comment]
Capture d’écran de 2020-07-09 11-02-27

Capture d’écran de 2020-07-09 11-02-29

Capture d’écran de 2020-07-09 11-02-30

Capture d’écran de 2020-07-09 11-02-31

Capture d’écran de 2020-07-09 11-02-33

Capture d’écran de 2020-07-09 11-02-35

Capture d’écran de 2020-07-09 11-02-37

Capture d’écran de 2020-07-09 11-02-39

Capture d’écran de 2020-07-09 11-02-40

Capture d’écran de 2020-07-09 11-02-42

Capture d’écran de 2020-07-09 11-02-45

Capture d’écran de 2020-07-09 11-02-47

Capture d’écran de 2020-07-09 11-02-48

Capture d’écran de 2020-07-09 11-02-50

Capture d’écran de 2020-07-09 11-02-52

**Comment by chibenwa** [comment]
I'm merging this skeleton to be moving things forward, however reviews are still welcome.

**Comment by chibenwa** [comment]
Rattached to James-3302 & merged

**Pull request closed without merging**